File: core-commons/src/main/java/net/blueshell/common/dto/BlogDTO.java
package net.blueshell.common.dto;


import lombok.*;
import net.blueshell.common.Image;

import java.sql.Timestamp;
import java.util.List;

@Data
@EqualsAndHashCode(callSuper = true)
public class BlogDTO extends BaseDTO {
    private String id;
    private String title;
    private String html;
    private Timestamp publishedAt;
}

File: api/src/main/kotlin/net/blueshell/api/blog/web/dto/BlogDTO.kt
package net.blueshell.api.blog.web.dto

import io.swagger.v3.oas.annotations.media.Schema
import jakarta.validation.constraints.NotBlank
import net.blueshell.api.shared.dto.AuditedAutoIdDTO
import java.time.Instant

@Schema(name = "Blog")
data class BlogDTO(
    var url: String? = null,

    @field:NotBlank
    var title: String,

    @field:NotBlank
    var html: String,

    var publishedAt: Instant
) : AuditedAutoIdDTO()
